src/data/variationOfCycleChecker.ts

```diff
@@ -0,0 +1,72 @@
+import {DbTricksTableZod} from "../lib/database/schemas/CurrentVersionSchema"
+import { z } from "zod";
+
+type Trick = z.infer<typeof DbTricksTableZod>
+
+/**
+ * Tricks are set up in a way where variationOf defines a "parent" trick they belong to. 
+ * Important note here is that we do not have trees as a trick can have multiple parents.
+ * 
+ * As such, the set of tricks can be described as a set of disjunct directed cycle-free graphs.
```

Comment:
This might be a nitpick but are we sure that this graph will always be cycle free? We could enforce it without much of a penalty in the usability but I don't think that it is entirely impossible to run into a situation where there is a cyclic connection. Consider the following simplified idea:

Trick A is a parent to trick B because of the general movement
Trick B is a parent to trick C because of the now more difficult general movement
Trick C is a parent to trick A because of a strange hand movement

Therefore, even though unlikely, it is possible to have a cylce in the graph based on valid data.

Reply:
Or, on a much simpler level: There could be two tricks X and Y which are very similar to each other both in movement and skill level and are therefore marked as each others parents.

Reply:
That's a discussion that is out of scope of this PR. I would keep it as is for now and create a new PR in case we decide that cycles are fine.

Reply:
Makes sense but we should keep this in mind

Reply:
I feel like for now we can keep it as it is. As soon as we add tricks that create cycles we will notice it and then we can still discuss if it still makes sense